### PR TITLE
[Design bug] After using highlight to edit, the cursor is placed after the highlighted word

### PR DESCRIPTION
**Pull request by nambatee**
https://phabricator.wikimedia.org/T219521

### CODE REVIEW

File: Wikipedia/Code/SectionEditorViewController.swift [hunk 1]
```diff
@@ -346,31 +341,26 @@ extension SectionEditorViewController: UIScrollViewDelegate {
...
+                self.needsSelectLastSelection = true
+                vc.theme = self.theme
+                vc.section = self.section
+                vc.wikitext = wikitext
+                vc.delegate = self
+                vc.funnel = self.editFunnel
+                self.navigationController?.pushViewController(vc, animated: true)
             }
-        } else {
-            let message = WMFLocalizedString("wikitext-preview-changes-none", value: "No changes were made to be previewed.", comment: "Alert text shown if no changes were made to be previewed.")
```
Comment (joewalsh) [inline Wikipedia/Code/SectionEditorViewController.swift]: Seems like this should stay and at least do a string comparison to the original wikitext to verify that something was changed - currently it allows the user to go through the process and fails silently

Comment (nambatee) [reply Wikipedia/Code/SectionEditorViewController.swift]: the `progressButton` should be disabled if no changes were made - what steps do you take to go through and fail silently?

File: www/codemirror/codemirror-editTextSelection.js [hunk 2]
```diff
@@ -110,7 +110,7 @@ const getWikitextRangeToSelect = (wikitextBeforeSelection, wikitextSelection) =>
 }
 
 const scrollToAndHighlightRange = (range, codemirror) => {
-  codemirror.setCursor(range.to)
+  codemirror.setSelection(range.from, range.to, {scroll: false})
 
   let marker = null
   marker = codemirror.markText(range.from, range.to, {
```
Comment (joewalsh) [inline www/codemirror/codemirror-editTextSelection.js]: `markText` can be removed since the text is highlighted by the selection now

Comment (nambatee) [reply www/codemirror/codemirror-editTextSelection.js]: makes sense to me but given https://phabricator.wikimedia.org/T215448#5012000 - maybe we can remove it after confirming with Carolyn? I added a comment to the ticket https://phabricator.wikimedia.org/T219521#5075653

**Pull request merged**